driver/src/main/java/org/neo4j/driver/internal/pool/StandardConnectionPool.java

```diff
@@ -71,25 +65,19 @@
      */
     private final ValidationStrategy<PooledConnection> connectionValidation;
 
-    private final int connectionsPerDatabase;
     private final Clock clock;
+    private final Config config;
 
-    public StandardConnectionPool( Logging logging )
-    {
-        this( loadConnectors( logging ) );
-    }
-
-    public StandardConnectionPool( Collection<Connector> conns )
+    public StandardConnectionPool( Config config )
```

Comment:
Same thing: creation of class B doesn;t belong in class A precisely because of what you said: it is not the responsibility of A to know about creation of class B. In particular wrt config, but in general also